packages/sit-onyx/src/components/OnyxBadge/OnyxBadge.vue

```diff
@@ -0,0 +1,94 @@
+<script lang="ts" setup>
+import { useDensity } from "../../composables/density";
+import type { OnyxBadgeProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxBadgeProps>(), {
+  variation: "primary",
+});
+
+const { densityClass } = useDensity(props);
+
+defineSlots<{
+  /**
+   * Badge content.
+   */
+  default(): unknown;
+}>();
+</script>
+
+<template>
+  <div class="onyx-badge" :class="[`onyx-badge--${props.variation}`, densityClass]">
+    <div class="onyx-badge--content onyx-truncation-ellipsis">
```

Comment:
BEM says this is an element, not a modifyer so you should use
```suggestion
    <div class="onyx-badge__content onyx-truncation-ellipsis">
```